### PR TITLE
feat: Create read only RichTextField to display formatted content

### PR DESCRIPTION
**Pull request by DeanGilewicz**
- https://app.clubhouse.io/homebound-team/story/6303/create-read-only-richtextfield-to-display-formatted-content
- add `readOnly` prop to `RichTextField` to render a `div` with formatted content
- add `dompurify` to sanitize read only content
- add `readOnly` prop to `BoundRichTextField` as a passthrough to `RichTextField`
- add storybook test for read only `RichTextField`

### CODE REVIEW

**Review by stephenh — APPROVED** [review]
Would like to sanity check the `readOnly` flipping true/false/true/false/etc but otherwise looks great!

`package-lock.json` snuck in, fwiw for old/legacy reasons this repo uses yarn/yarn.lock; we could switch it over, just haven't gotten to it.

File: src/components/RichTextField.tsx [hunk 1]
```diff
@@ -26,6 +27,8 @@ export interface RichTextFieldProps {
   onBlur: () => void;
   /** Called when the component is in focus. */
   onFocus: () => void;
+  /** For rendering formatted text */
+  readOnly?: boolean | undefined;
```
[the remark this answers is not in the document]
Comment (DeanGilewicz) [reply src/components/RichTextField.tsx:31]: makes sense - will update. Thanks

File: src/components/RichTextField.tsx [hunk 2]
```diff
@@ -62,88 +65,103 @@ export function RichTextField(props: RichTextFieldProps) {
...
-      if (!editor.current) {
-        throw new Error("editor not found");
-      }
-      if (mergeTags !== undefined) {
-        attachTributeJs(mergeTags, editorElement!);
-      }
+      if (!readOnly) {
+        const editorElement = document.getElementById(`editor-${id}`);
+        if (!editorElement) {
+          throw new Error("editorElement not found IDIOT");
```
[the remark this answers is not in the document]
Comment (DeanGilewicz) [reply src/components/RichTextField.tsx:71]: 🤦 thanks

File: src/components/RichTextField.tsx [hunk 3]
```diff
@@ -62,88 +65,103 @@ export function RichTextField(props: RichTextFieldProps) {
...
-      }
-
-      editor.current = (editorElement as any).editor;
-      if (!editor.current) {
-        throw new Error("editor not found");
-      }
-      if (mergeTags !== undefined) {
-        attachTributeJs(mergeTags, editorElement!);
-      }
+      if (!readOnly) {
```
[the remark this answers is not in the document]
Comment (DeanGilewicz) [reply src/components/RichTextField.tsx:68]: Yeah happy to update - I've seen either way although I agree it's cleaner to return early

File: src/components/RichTextField.tsx [hunk 4]
```diff
@@ -62,88 +65,103 @@ export function RichTextField(props: RichTextFieldProps) {
...
+          editorElement.removeEventListener("trix-change", trixChange as any);
+          editorElement.removeEventListener("trix-blur", trixBlur as any);
+          editorElement.removeEventListener("trix-focus", trixFocus as any);
+          window.removeEventListener("trix-file-accept", preventDefault);
+        };
+      }
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [],
+    [readOnly],
```
[the remark this answers is not in the document]
Comment (DeanGilewicz) [reply src/components/RichTextField.tsx:123]: no worries - it just works :) but will add a storybook story to showcase this

File: src/forms/BoundRichTextField.tsx [hunk 5]
```diff
@@ -8,11 +8,13 @@ export type BoundRichTextFieldProps = Omit<RichTextFieldProps, "value" | "onChan
   field: FieldState<any, string | null | undefined>;
   // Optional in case the page wants extra behavior
   onChange?: (value: string | undefined) => void;
+  // Optional for displaying formatted text
+  readOnly?: boolean | undefined;
```
[the remark this answers is not in the document]
Comment (DeanGilewicz) [reply src/forms/BoundRichTextField.tsx:12]: ah yes, good catch

**Pull request merged**